sky/backends/cloud_vm_ray_backend.py

```
@@ -2276,6 +2290,14 @@ def is_provided_ips_valid(
 
             use_internal_ips = self._use_internal_ips()
 
+            # Currently VPN is not supported.
+            # TODO(yi): move this warning earlier, or consider to move
+            # VPN into the features.
+            if vpn_config is not None:
+                logger.warning('VPN is not supported for clouds that do not '
+                               'support the new provisioner API. Ignoring the '
+                               'provided VPN configuration.')
```

Lets raise an error before any provision happens.